qrest/conf.py

```diff
@@ -77,22 +84,51 @@ def _validate(self):
             raise RestClientConfigurationError('parameter "multiple" must be boolean')
         if not isinstance(self.description, str):
             raise RestClientConfigurationError('parameter "description" must be string')
+
         if self.exclusion_group:
             if not isinstance(self.exclusion_group, str):
                 raise RestClientConfigurationError("group name must be a string")
             if not self.exclusion_group.strip():
                 raise RestClientConfigurationError("group name must be a string")
+
         if self.default and self.required:
             raise RestClientConfigurationError(
                 "you cannot combine required=True and a default setting"
             )
-        if self.choices and not isinstance(self.choices, list):
-            raise RestClientConfigurationError("choices must be a list")
-        if self.default and self.choices:
-            if self.default not in self.choices:
-                raise RestClientConfigurationError(
-                    "if there is a choices list, default must be in this list"
-                )
+
+        if self.choices:
```

Comment:
If `self.choices` is the empty list, then `self.choices` evaluates to `False` and lines 100-109 aren't executed. This might be an edge case, but
``` python
if self.choices:
```
is not the same as
``` python
if self.choices is not None:
```

EDIT: I see that in other locations too, e.g. line 194 of this file. I've made a comment at that line.